docs/derived-variables.qmd

```diff
@@ -0,0 +1,190 @@
+---
+title: Derived Variables Scope 
+---
+
+## Objective
+
+The objective of this document is to define the scope and specifications for 
+derived variables in the library.
+
+## Context
+
+The variable details sheet within the recodeflow library enables a user
+to encode the rules to create a new variable from one or more starting variables.
+Broadly, any rule that maps the values of one variable into another can be encoded in the variable details sheet. Examples include rules that 
+map one or more categories of a variable into a new category or a rule that 
+maps an interval of a continuous variable into a category. Variables that cannot be created with simple mapping rules can be recoded by using a 
+function. Examples include variables that require mathematical operations  
+or conditional logic.
+
+Any variable that requires a function for its creation is considered a derived variable. A good example of this is BMI which requires mathematical operations
+for its calculation and optionally conditional logic to validate its inputs. This document will go over the scope and specifications
+for such derived variables and the functions used to derive them.
+
+## Scope
+
+### Function Language
+
+The library should support only functions written in R. The R ecosystem is moving 
+towards allowing its users to execute code from other statistical languages. An 
+example is the [reticulate](https://rstudio.github.io/reticulate/index.html)
+package which enables the execution of arbitrary 
+Python code within a .R file. Example below,
+
+```{r, python-in-R}
+# Python code that creates a function to calculate BMI
```

Comment:
I suggest the qmd should be executable- meaning add the reticulate and knitr library to this file.

Reply:
I added them to the Suggests field in the DESCRIPTION file. Do you also mean to load them in using `library`?